improver_tests/utilities/test_extract_from_table.py

```diff
@@ -0,0 +1,99 @@
+# (C) Crown copyright, Met Office. All rights reserved.
+#
+# This file is part of IMPROVER and is released under a BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Tests of read_from_table utilities"""
+
+import numpy as np
+import pytest
+
+from improver.synthetic_data.set_up_test_cubes import set_up_variable_cube
+from improver.utilities.extract_from_table import ExtractValueFromTable
+
+
+@pytest.fixture
+def table():
+    """Set up a dictionary representing a table of data"""
+
+    table_dict = {
+        "data": {
+            0: {1: 0.95, 2: 0.9},
+            10.19: {1: 0.9, 2: 0.85},
+            15.43: {1: 0.85, 2: 0.8},
+        },
+        "metadata": {"name": "Gust Factor", "units": "1"},
+    }
+    return table_dict
+
+
+@pytest.fixture
+def lapse_class():
+    """Set up a cube containing lapse class"""
+    data = np.full(
+        (2, 2), 0, dtype=np.float32
+    )  # Values will be overwritten in the test
+    return set_up_variable_cube(data=data, name="lapse_class")
+
+
+@pytest.fixture
+def wind_gust_900m():
+    """Set up cube containing 900m wind gust data"""
+    data = np.full(
+        (2, 2), 0, dtype=np.float32
+    )  # Values will be overwritten in the test
+    return set_up_variable_cube(data=data, name="900m_wind_gust")
+
+
+@pytest.mark.parametrize("new_name", [None, "adjusted_wind_speed_of_gust"])
+@pytest.mark.parametrize(
+    "lapse_class_value, wind_gust_value, expected",
+    [
+        [1, -4, 0.95],
+        [1, 5, 0.95],
+        [2, 10.19, 0.85],
+        [2, 14, 0.85],
+        [2, 15.43, 0.8],
+        [3, 20, 0.8],
+    ],
+)
+def test_read_table(
+    table,
+    lapse_class,
+    wind_gust_900m,
+    lapse_class_value,
+    wind_gust_value,
+    expected,
+    new_name,
+):
+    """Test plugin to extract values from table"""
+    wind_gust_900m.data.fill(wind_gust_value)
+    lapse_class.data.fill(lapse_class_value)
+    result = ExtractValueFromTable(row_name="lapse_class", new_name=new_name)(
+        wind_gust_900m, lapse_class, table=table
+    )
+    expected_data = np.full_like(
+        lapse_class.data, fill_value=expected, dtype=np.float32
+    )
+    expected_cube = lapse_class.copy(data=expected_data)
+    expected_cube.units = "1"
+    if new_name:
+        expected_cube.rename(new_name)
+    np.testing.assert_array_almost_equal(result.data, expected_cube.data)
+    assert result == expected_cube
+
```

Comment:
There are quite a lot of things missing from this testing as it stands:
- disordered tables to test the dataframe index sorting
- data values that are within `tol` of the column or row index value to test the modification of the chosen index
- setting of the output units using the table metadata
- a 1-D table in the unit test would be easier to interpret than the acceptance test (but you may as well keep the acceptance test as well).